### PR TITLE
Plasma Stage 1 Review 

### PR DESCRIPTION
**Pull request by Dan-Nolan**
- Updates Tasks, Details and Completion Messaging

### CODE REVIEW

File: projects/Plasma MVP/JavaScript/Constructor Plasma/details.md [hunk 1]
```diff
@@ -4,14 +4,16 @@ This version of Plasma is created under the assumption of a single `operator`. T
...
-In this Plasma implementation we have two separate block numbers that will be stored in our contract, `currentPlasmaBlock` and `currentDepositBlock`. We also have a `BLOCK_BUFFER` constant of 1000. . 
+In this Plasma implementation we have two separate block numbers that will be stored in our contract: `currentPlasmaBlock` and `currentDepositBlock`. 
 
 Deposits are first submitted to the Plasma contract and then updated on the Plasma chain. As a result, a deposit block is immediately created on the Plasma chain once it is updated. 
 
 On the other hand, Plasma transactions occur first on the Plasma chain, then are written to the `currentBlock`, and finally, once the `currentBlock` is submitted by the `operator`, the Plasma contract is updated.
 
+## Block Buffer
+
 As you can see, we need a buffer between deposits and Plasma chain transaction blocks in order to keep the Plasma contract and Plasma chain in sync. If there wasn't a buffer, its quite possible that a submitted block could have the same block number as a deposit block.
```
Comment (Dan-Nolan) [inline projects/Plasma MVP/JavaScript/Constructor Plasma/details.md]: Also think we should probably clean this up a bit. I'm not sure my understanding is perfect at the moment, so I haven't reworded it.

File: projects/Plasma MVP/JavaScript/Constructor Plasma/task.md [hunk 2]
```diff
@@ -1,6 +1,10 @@
 ## Plasma Contract
 
-The first part in building a Plasma chain is creating the smart contract which will ultimately hold the funds available on the Plasma chain, allow users to deposit into the chain, allow users to exit from the chain, and allow users to challenge any exits from the chain.
+Let's create our Plasma Contract! 
+
+First we'll create the smart contract which will ultimately hold the funds available on the Plasma chain. It will allow users to deposit into the chain, allow users to exit from the chain, and allow users to challenge any exits from the chain.
```
Comment (Roots2D) [inline projects/Plasma MVP/JavaScript/Constructor Plasma/task.md]: Should also mention finalizing exits

Comment (Roots2D) [inline projects/Plasma MVP/JavaScript/Constructor Plasma/task.md]: Should also mention finalizing exits

File: projects/Plasma MVP/JavaScript/Constructor Plasma/task.md [hunk 3]
```diff
@@ -1,6 +1,10 @@
...
+
+You'll need to define four variables. One for this `address` of the [operator](?tab=details&scroll=Single%20Operator), a `uint` for both of the [Plasma Blocks](?tab=details&scroll=Plasma%20Blocks), and a `uint` for [Block Buffer](?tab=details&scroll=Block%20Buffer).
```
Comment (Roots2D) [inline projects/Plasma MVP/JavaScript/Constructor Plasma/task.md]:
```suggestion
You'll need to define four variables. One for the `address` of the [operator](?tab=details&scroll=Single%20Operator), a `uint` for both of the [Plasma Blocks](?tab=details&scroll=Plasma%20Blocks), and a `uint` for [Block Buffer](?tab=details&scroll=Block%20Buffer).
```

Comment (Roots2D) [inline projects/Plasma MVP/JavaScript/Constructor Plasma/task.md]:
```suggestion
You'll need to define four variables. One for the `address` of the [operator](?tab=details&scroll=Single%20Operator), a `uint` for both of the [Plasma Blocks](?tab=details&scroll=Plasma%20Blocks), and a `uint` for the [Block Buffer](?tab=details&scroll=Block%20Buffer).
```

File: projects/Plasma MVP/JavaScript/Constructor Plasma/details.md [hunk 4]
```diff
@@ -4,14 +4,16 @@ This version of Plasma is created under the assumption of a single `operator`. T
...
 
- ## Plasma Blocks and Buffer
+## Plasma Blocks 
 
-In this Plasma implementation we have two separate block numbers that will be stored in our contract, `currentPlasmaBlock` and `currentDepositBlock`. We also have a `BLOCK_BUFFER` constant of 1000. . 
+In this Plasma implementation we have two separate block numbers that will be stored in our contract: `currentPlasmaBlock` and `currentDepositBlock`. 
 
 Deposits are first submitted to the Plasma contract and then updated on the Plasma chain. As a result, a deposit block is immediately created on the Plasma chain once it is updated. 
 
 On the other hand, Plasma transactions occur first on the Plasma chain, then are written to the `currentBlock`, and finally, once the `currentBlock` is submitted by the `operator`, the Plasma contract is updated.
```
Comment (Roots2D) [inline projects/Plasma MVP/JavaScript/Constructor Plasma/details.md]:
```suggestion
On the other hand, Plasma transactions first occur on the Plasma chain. These transactions are immediately written to the `currentBlock` on the Plasma chain. Once the `operator` submits the `currentBlock` from the Plasma chain to the Plasma contract, both are in sync with one another. ```

Comment (Dan-Nolan) [reply projects/Plasma MVP/JavaScript/Constructor Plasma/details.md]: Bit hard to tell, but should the three backticks be there at the end of this change?

File: projects/Plasma MVP/JavaScript/Constructor Plasma/details.md (hunk 1 above)
Comment (Roots2D) [inline projects/Plasma MVP/JavaScript/Constructor Plasma/details.md]:
```suggestion
The `BLOCK_BUFFER` serves as a tool to keep the Plasma contract and Plasma chain in sync. As described above, we have to keep track of deposit blocks, which begin on the Plasma contract, and Plasma transaction blocks, which begin on the Plasma chain. Therefore, we start our deposit block at 1 and our Plasma transaction block at 1000 (our `BLOCK_BUFFER`) and increment by 1 and 1000 respectively. In this way, we can create up to 999 deposits before interfering with the transactions occurring on the Plasma chain. 

You will be exposed concept multiple times throughout the tutorial which should help to solidify the idea behind the framework.
```

Comment (Roots2D) [reply projects/Plasma MVP/JavaScript/Constructor Plasma/details.md]: Added a revised version. Let me know if this clears it up.

File: projects/Plasma MVP/JavaScript/Constructor Plasma/details.md (hunk 4 above)
Comment (Roots2D) [reply projects/Plasma MVP/JavaScript/Constructor Plasma/details.md]: fixed. Was part of the suggestion annotation

Comment (Dan-Nolan) [reply projects/Plasma MVP/JavaScript/Constructor Plasma/details.md]: Awesome

**Pull request merged**